Mathlib/Dynamics/Circle/MinimalRotation.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Data.Real.Irrational
+import Mathlib.Topology.Instances.AddCircle
+
+/-!
+# Irrational rotation is minimal
+
+In this file we prove that the multiples of an irrational element of an `AddCircle` are dense.
+-/
+
+open Set Filter
+open scoped Pointwise Topology
+
+section
+
+variable {G : Type*} [LinearOrderedAddCommGroup G] [DenselyOrdered G] [Archimedean G] [Nontrivial G]
+    [TopologicalSpace G] [OrderTopology G]
+
+theorem AddSubgroup.dense_xor'_cyclic (s : AddSubgroup G) :
+    Xor' (Dense (s : Set G)) (∃ a, s = .zmultiples a) := by
```

Comment:
Why not something like this?
```suggestion
    (Dense (s : Set G)) ↔ ¬ (∃ a, s = .zmultiples a) := by
```
You can even push the negation if you want.

Reply:
That's exactly the next lemma. I don't remember why I wanted to have `Xor'` here. I'll look at this again before removing "Draft" status from the PR.

Reply:
UPD: I chose `Xor'` because it's symmetric.